plugins/search-backend-module-elasticsearch/src/engines/ElasticSearchSearchEngine.ts

```diff
@@ -266,6 +303,20 @@ export class ElasticSearchSearchEngine implements SearchEngine {
       }
     });
 
+    indexer.on('finish', async () => {
```

Comment:
My understanding is that the `finish` event fires each time an indexing run completes (for each and every configured collator).

Reading [the ES docs on index templates](https://www.elastic.co/guide/en/elasticsearch/reference/current/index-templates.html), I'm seeing...

> _An index template is a way to tell Elasticsearch how to configure an index when it is created. For data streams, the index template configures the stream’s backing indices as they are created. Templates are configured prior to index creation._

I'm wondering if we want to call `putIndexTemplate` less frequently and much earlier in the process, perhaps even only once at startup?

Reply:
@iamEAP Good point, I wonder in that case if `setIndexTemplate` should be directly responsible for calling `putIndexTemplate`? Because we want to be able to instantiate the search engine before we _eventually_ set the custom template?

Reply:
I like it!

Reply:
Updated!